ipa-core/src/ff/prime_field.rs

```diff
@@ -165,10 +166,7 @@ macro_rules! field_impl {
             fn mul(self, rhs: Self) -> Self::Output {
                 debug_assert!(<$store>::try_from(Self::PRIME).is_ok());
                 // check container for multiply is large enough
-                debug_assert!(
-                    <$store_multiply>::MAX >> $bits
-                        >= <$store_multiply>::try_from(<$store>::MAX).unwrap()
-                );
+                const_assert!((<$store_multiply>::MAX >> $bits) as u128 >= (<$store>::MAX) as u128);
```

Comment:
hopefully the compiler will optimize it, but you could also lift this statement outside of this function because it belongs to the macro itself